poetry/utils/env.py

```diff
@@ -1327,7 +1329,7 @@ def _bin(self, bin: str) -> str:
 
         return str(bin_path)
 
-    def __eq__(self, other: "Env") -> bool:
+    def __eq__(self, other: object) -> bool:
```

Comment:
this is required to satisfy Liskov subsitution principle- namely that the derived class's `__eq__` should have the same, or a less restrictive function signature for any overridden baseclass methods.